statsmodels/base/elastic_net.py

```diff
@@ -236,6 +237,12 @@ def fit_elasticnet(model, method="coord_descent", maxiter=100,
     else:
         scale = 1.
 
+    # The degrees of freedom should reflect the number of parameters
+    # in the refit model, not including the zeros that are displayed
+    # to indicate which variables were dropped.
+    model.df_model = len(ii)
+    model.df_resid = model.nobs - model.df_model
```

Comment:
In the models we exclude the constant in df_model, regression models subtract has_const, GLM subtracts 1
self.df_model = np_matrix_rank(self.exog) - 1
and then we correct again in df_resid to compensate

Also, the df_xxx should be changed in the results instance and not in the model instance. AFAICS, we can repeatedly call fit with different penalization weights which would change the df in the model.
(However, I think this is currently not a bug because the results instance should copy the df, and be independent of any incorrect model.df.)

Reply:
@josef-pkt not sure I follow this, can you give more explicit guidance about what to change?

Reply:
My comment was a fast response, and I need to look more closely.

about the second: `model` is the original model, at the end it is used to create a new results instance with it.
However, df_resid is changed in the original `model` which now doesn't have the df corresponding to the original model but corresponding to the latest `fit`, i.e. the `model` has incorrect or "stale" state
e.g. #3949  changed it for the old discrete fit_regularized
jbrockmendel and I had lengthy discussions, A FAQ-D #1723 contains mainly the links to issues.

I have to check df_model because we don't treat has_const the same way in all models. Maybe the best would be
something like
df_model = model.df_model - <number of params regularized to 0>
which wouldn't change the treatment of df_model (except for the unlikely case when the constant is regularized to zero)

Reply:
df_model = model.df_model - <number of params regularized to 0>
would have as side effect that the resulting df_model is wrong if `model.df_model` has been changed by a previous fit instead of remaining at the original.

(I haven't worked my way through the entire fit_regularized code path, so this is still a generic comment.)

Reply:
a possible quickfix, hackish solution to avoiding wrong model.df_resid:
It is difficult to set the df_resid properly on a results instance for the general case.
So one possibility would be to keep the current code changing model.df_model and model.df_resid before creating the results instance, but then setting it back after the results instance has been created, using temporary storage of the old values.
In the longer term we need a good solution, instead of the inconsistent way we currently handle this across models.